### ISSUE TITLE
Somewhat confusing error with `extended_key_value_attributes`

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#85066]
### Code

Note that `../README.md` must **not** exist in order to reproduce this error:

```rust
#![feature(extended_key_value_attributes)]
#![doc = include_str!("../README.md")]
```

[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=7ead434499cf146866df4672d2d19ef4)

### Output

The first error is what I would expect, but the second is a bit confusing and weird:

```
error: couldn't read src/../README.md: No such file or directory (os error 2)
 --> src/lib.rs:2:10
  |
2 | #![doc = include_str!("../README.md")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected token: `(/*ERROR*/)`
 --> src/lib.rs:2:10
  |
2 | #![doc = include_str!("../README.md")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

Perhaps instead of emitting ``unexpected token: `(/*ERROR*/)` `` when an error token is encountered, we could use a `delay_span_bug` since an error should have already occurred.

If for some reason an error had *not* already been emitted, the compiler would crash; otherwise, it would just show the original error.

_Originally posted by @camelid in https://github.com/rust-lang/rust/issues/83366#issuecomment-834750746_

**Comment by camelid** [comment]
Note that `extended_key_value_attributes` is stable now (though it hasn't reached the stable channel yet), so this no longer requires a feature flag.

**Comment by jyn514** [comment]
Mentoring instructions: in https://github.com/rust-lang/rust/blob/34327f6eee06835f81a26765a70ad51d44343ba3/compiler/rustc_parse/src/parser/expr.rs#L1496, check if the token is a `token::Interpolated(token::NtExpr(expr) | NtLiteral(expr))` and the expression is an error; if so, use `delay_span_bug` instead of `struct_span_err`.

**Comment by Wardenfar** [comment]
Hi @jyn514 
I was searching for a first issue and i have a question :
What does ```delay_span_bug``` ? Because the function that you pin must return a ```PResult``` but ```delay_span_bug``` has return type ```()```
Sorry if i misread something

**Comment by jyn514** [comment]
@Wardenfar delay_span_bug means "if the compiler finishes running without first reporting another error, panic". It's useful here because error tokens should always report in an error being reported.

I think anything you return is probably fine as long as it doesn't cause further cascading errors :)

**Comment by Wardenfar** [comment]
@jyn514 i fixed it but for adding a test case : i run into an edge case
the output message of the macro : include_str! is not normalized with $DIR
the ui test is working only for me. Have you a tip or i have to create an issue ?
Thanks

````
#![feature(extended_key_value_attributes)]
#![doc = include_str!("../not_existing_file.md")]
struct Documented {}
//~^^ ERROR 2:10: 2:49: couldn't read /home/<username>/dev/rustc/rust/src/test/ui/attributes/../not_existing_file.md: No such file or directory (os error 2)

fn main() {}
```

**Comment by jyn514** [comment]
@Wardenfar you can use `normalize-stderr-test`: https://rustc-dev-guide.rust-lang.org/tests/adding.html#output-normalization. But I'm surprised $DIR doesn't work :/ that does seem like a bug.

**Comment by Wardenfar** [comment]
Sorry I was not clear : the $DIR is working for the test.stderr 
But the path is not replace for the error comment in the test.rs because the path read by include_str is in the error message
@jyn514

**Comment by jyn514** [comment]
@Wardenfar don't put the whole name in the .rs file - you just need to put a substring. Something like `//~^^ ERROR No such file or directory`.
